devices/flash-flexspi/nor/nor.c

```diff
@@ -178,28 +178,59 @@ __attribute__((section(".noxip"))) int nor_eraseSector(flexspi_t *fspi, u8 port,
 }
 
 
-__attribute__((section(".noxip"))) int nor_eraseChip(flexspi_t *fspi, u8 port, time_t timeout)
+__attribute__((section(".noxip"))) static int nor_mode4ByteAddr(flexspi_t *fspi, u8 port, int en4b, time_t timeout)
 {
 	struct xferOp xfer;
-
-	int res = nor_writeEnable(fspi, port, 1, timeout);
-	if (res < EOK) {
-		return res;
-	}
+	int seqCode = ((en4b == 0) ? fspi_exit4byteAddr : fspi_enter4byteAddr);
 
 	xfer.op = xfer_opCommand;
 	xfer.port = port;
 	xfer.timeout = timeout;
 	xfer.addr = 0;
-	xfer.seqIdx = LUT_SEQIDX(fspi_eraseChip);
-	xfer.seqNum = LUT_SEQNUM(fspi_eraseChip);
+	xfer.seqIdx = LUT_SEQIDX(seqCode);
+	xfer.seqNum = LUT_SEQNUM(seqCode);
 
-	res = flexspi_xferExec(fspi, &xfer);
-	if (res < EOK) {
-		return res;
+	return flexspi_xferExec(fspi, &xfer);
+}
+
+
+__attribute__((section(".noxip"))) int nor_eraseChipDie(flexspi_t *fspi, u8 port, u32 capFlags, int dieCount, size_t dieSize, time_t timeout)
+{
+	struct xferOp xfer;
+	int dieIndex, res = -EIO;
+
+	for (dieIndex = 0; dieIndex < dieCount; ++dieIndex) {
+		res = nor_writeEnable(fspi, port, 1, timeout);
+		if (res < EOK) {
+			break;
+		}
+
+		if ((dieCount > 1) && ((capFlags & NOR_CAPS_EN4B) != 0u)) {
+			res = nor_mode4ByteAddr(fspi, port, 1, timeout);
+			if (res < EOK) {
+				return res;
```

Comment:
I'd use either break or change other breaks to return to be coherent

Reply:
Done.